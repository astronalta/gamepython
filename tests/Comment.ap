Comment < Object {
# This is a class comment for the current class scope.  It can span multiple
# lines, and all whitespace will be stripped from the beginning of the 
# comment block.

    comment1() {
        # This is a function comment for the current function scope.  It can 
        # span multiple lines, and all whitespace will be stripped from the 
        # beginning of the comment block. 
        #
        # All comments until the end of the function or the first non-comment
        # symbol are included.

        if true {
        
        } # Comment in a strange place
    } # Comment in a strange place

    comment2() {
        # The quick brown fox jumped over the lazy dog.
    }

    x = String()
    # Another comment in a strange place
    

}
