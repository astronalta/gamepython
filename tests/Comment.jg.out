Module
  name: 
  feature0: Class
    name: Comment
    comment:
      This is a class comment for the current class scope.  It can span
      multiple lines, and all whitespace will be stripped from the beginning of
      the  comment block. 
    feature0: Function
      name: comment1
      type: Void
      comment:
        This is a function comment for the current function scope.  It can
        span multiple lines, and all whitespace will be stripped from the
        beginning of the comment block.
        
        All comments until the end of the function or the first non-comment
        symbol are included. 
      formal0: Formal
        name: self
        type: Comment
      block: Block
        child0: Conditional
          guard: BooleanLiteral(1)
          true: Block
    feature1: Function
      name: comment2
      type: Void
      comment:
        The quick brown fox jumped over the lazy dog. 
      formal0: Formal
        name: self
        type: Comment
      block: Block
    feature2: Attribute
      name: x
      type: String
      initializer: Construct
        type: String
    feature3: Function
      name: @destroy
      type: Void
      formal0: Formal
        name: self
        type: Comment
      block: Block
    feature4: Function
      name: @init
      type: Void
      block: Block
    feature5: Function
      name: x?
      type: String
      formal0: Formal
        name: self
        type: Comment
      block: Block
        child0: Return
          expression: IdentifierRef
            name: x
